tests/messengers/common/test_messenger_interface.py

```diff
@@ -0,0 +1,228 @@
+            self.received_updates = []
+
+        # First worker is ordered to hand over control to (0, id: 2), which immediately gives it back because of the ContextLock(). Then, id: 3 gets completed right away, due to there not being any more 'await' statements there.
+        # What's important here is that the worker for the "id: 2" couldn't start working on it's task immediately, because of ContextLock(), proving it's function. Even though it was given control with an 'await' just below.
+        async def _process_request(self, ctx_id, update: Message, pipeline: Pipeline):
+            context = await pipeline._run_pipeline(update, ctx_id)
+            if context.last_response.text == "id: 1":
+                await asyncio.sleep(0)
+            await self._respond(ctx_id, context.last_response)
+
+    new_pipeline = Pipeline.from_script(
+        ECHO_SCRIPT,
+        start_label=("echo_flow", "start_node"),
+        fallback_label=("echo_flow", "start_node"),
+        messenger_interface=TestPollingInterface(),
+    )
+    interface = new_pipeline.messenger_interface
+    asyncio.run(new_pipeline.messenger_interface.run_in_foreground(new_pipeline, loop=interface.not_obtained_updates))
+    for i in range(3):
+        assert interface.expected_updates[i][1] == interface.received_updates[i]
+    assert interface.not_obtained_updates() == False
+
+
+def test_worker_shielding():
+    class TestPollingInterface(BaseTestPollingInterface):
+        def __init__(self):
+            super().__init__()
+            self.ctx_id = uuid.uuid4()
+            self.requests = ["1sa", "nb2", "3fdg", "g46", "2sh2", "h31", "m56", "72ds"]
+            self.requests.extend(self.requests)
+            self.requests.extend(self.requests)
+            self.requests.extend(self.requests)
+            self.requests.extend(self.requests)
+            self.expected_updates = self.requests.copy()
+            self.received_updates = []
+            self.obtained_requests = False
+            self.requests_count = 0
+
+        def not_obtained_requests(self):
+            if len(self.requests) == 0:
+                self.obtained_requests = True
+                print("hi")
+                # Is this the right way to run a random async function?
+                asyncio.create_task(self.shutdown())
+
+                # This shuts down the interface via the main method. The asyncio.CancelledError won't affect the workers due to shielding. If all the messages still get processed, then the workers are shielded.
+            return True
+
+        async def _get_updates(self):
+            if len(self.requests) > 0:
+                self.requests_count += 1
+                request = self.requests.pop(0)
+                return [(self.ctx_id, Message(text=str(request)))]
+
+        async def _process_request(self, ctx_id, update: Message, pipeline: Pipeline):
+            while self.running:
+                await asyncio.sleep(0.05)
```

Comment:
Add comments on the purpose of these changes to the methods (what they do for the test).